src/lib/swapper/swappers/ThorchainSwapper/utils/makeMemoWithShortenedFinalAssetAmount.test.ts

```diff
@@ -0,0 +1,77 @@
+  it('should be 80 bytes if maxMemoSize is 80', () => {
+    const quotedMemo = `=:e:${RECEIVE_ADDRESS}:${expectedL1AmountOut}:ss:${affiliateBps}:${AGGREGATOR_TWO_LAST_CHARS}:${SHORTENED_FINAL_ASSET_ADDRESS}`
+
+    const modifiedMemo = makeMemoWithShortenedFinalAssetAmount({
+      maxMemoSize: 80,
+      memoWithoutMinAmountOut: quotedMemo,
+      finalAssetLimitWithManualSlippage: '950875902132134123',
+    })
+
+    expect(modifiedMemo.length).toBe(80)
+    expect(modifiedMemo).toBe(
+      `=${MEMO_PART_DELIMITER}e${MEMO_PART_DELIMITER}${RECEIVE_ADDRESS}${MEMO_PART_DELIMITER}${expectedL1AmountOut}${MEMO_PART_DELIMITER}ss${MEMO_PART_DELIMITER}${affiliateBps}${MEMO_PART_DELIMITER}${AGGREGATOR_TWO_LAST_CHARS}${MEMO_PART_DELIMITER}${SHORTENED_FINAL_ASSET_ADDRESS}${MEMO_PART_DELIMITER}95087590209`,
+    )
+  })
+
+  it('should be any bytes if maxMemoSize is Infinity', () => {
+    const quotedMemo = `=:e:${RECEIVE_ADDRESS}:${expectedL1AmountOut}:ss:${affiliateBps}:${AGGREGATOR_TWO_LAST_CHARS}:${SHORTENED_FINAL_ASSET_ADDRESS}`
+
+    const modifiedMemo = makeMemoWithShortenedFinalAssetAmount({
+      maxMemoSize: Infinity,
+      memoWithoutMinAmountOut: quotedMemo,
+      finalAssetLimitWithManualSlippage: '950875902132134123',
+    })
+
+    expect(modifiedMemo).toBe(
+      `=${MEMO_PART_DELIMITER}e${MEMO_PART_DELIMITER}${RECEIVE_ADDRESS}${MEMO_PART_DELIMITER}${expectedL1AmountOut}${MEMO_PART_DELIMITER}ss${MEMO_PART_DELIMITER}${affiliateBps}${MEMO_PART_DELIMITER}${AGGREGATOR_TWO_LAST_CHARS}${MEMO_PART_DELIMITER}${SHORTENED_FINAL_ASSET_ADDRESS}${MEMO_PART_DELIMITER}9508759021321341201`,
+    )
+  })
+
+  it('should be under 220 bytes if maxMemoSize is 220 and memo length is close to 220', () => {
+    // 210 bytes
+    const quotedMemo = `=:e:${BIG_ADDRESS}:${expectedL1AmountOut}:ss:${affiliateBps}:${AGGREGATOR_TWO_LAST_CHARS}:${SHORTENED_FINAL_ASSET_ADDRESS}`
+
+    const modifiedMemo = makeMemoWithShortenedFinalAssetAmount({
+      maxMemoSize: 220,
+      memoWithoutMinAmountOut: quotedMemo,
+      finalAssetLimitWithManualSlippage: '950875902132134123',
+    })
+
+    expect(modifiedMemo.length).toBeLessThanOrEqual(220)
+    expect(modifiedMemo).toBe(
+      `=${MEMO_PART_DELIMITER}e${MEMO_PART_DELIMITER}${BIG_ADDRESS}${MEMO_PART_DELIMITER}${expectedL1AmountOut}${MEMO_PART_DELIMITER}ss${MEMO_PART_DELIMITER}${affiliateBps}${MEMO_PART_DELIMITER}${AGGREGATOR_TWO_LAST_CHARS}${MEMO_PART_DELIMITER}${SHORTENED_FINAL_ASSET_ADDRESS}${MEMO_PART_DELIMITER}950875902132105`,
+    )
+  })
+
+  it('should be throwing if length cant be achieved', () => {
+    const quotedMemo = `=:e:${REALLY_BIG_ADDRESS}:${expectedL1AmountOut}:ss:${affiliateBps}:${AGGREGATOR_TWO_LAST_CHARS}:${SHORTENED_FINAL_ASSET_ADDRESS}`
+
+    expect(() =>
+      makeMemoWithShortenedFinalAssetAmount({
+        maxMemoSize: 220,
+        memoWithoutMinAmountOut: quotedMemo,
+        finalAssetLimitWithManualSlippage: '950875902132134123',
+      }),
+    ).toThrow()
```

Comment:
Also here possibly check error message

Reply:
Added the error message we suspect to have in this case, most of the time the exponent is just bigger than it's supposed to be